src/EdgeEval/constraint_applyer.rs

```diff
@@ -1,211 +1,655 @@
+        ArithExpression::Multiplication(
+            Box::new(ArithExpression::Int(3)),
+            Box::new(ArithExpression::Int(3)),
+        ),
+        ArithExpression::Clock(10),
+    );
+    let right = ArithExpression::ADif(ArithExpression::Int(10), ArithExpression::Clock(10));
+    //Testing: left < right
+    assert_eq!(get_indices(&left, &right, &decl), Ok((10, 10, 1)));
 }
 
 /// Assumes that the constraint is of the form left <?= right
 fn get_indices(
-    left: &BoolExpression,
-    right: &BoolExpression,
-    d: &component::Declarations,
-) -> (u32, u32, i32) {
-    let result = if let BoolExpression::Difference(i, j) = left {
-        try_form_index(get_clock(i, d), get_clock(j, d), get_constant(right))
-    } else if let Some(c) = get_constant(left) {
-        if let BoolExpression::Difference(i, j) = right {
-            try_form_index(get_clock(j, d), get_clock(i, d), Some(-c))
-        } else {
-            try_form_index(Some(0), get_clock(right, d), Some(-c))
+    left: &ArithExpression,
+    right: &ArithExpression,
+    d: &Declarations,
+) -> Result<(u32, u32, i32), String> {
+    let left = &(replace_vars(left, d).simplify())?;
+    let right = &(replace_vars(right, d).simplify())?;
+    let (clocks_left, clocks_right) = (left.clock_var_count(), right.clock_var_count());
+
+    if clocks_left + clocks_right > 2 {
+        return Err(String::from("Too many clocks"));
+    }
+
+    let (left_const, right_const) = (get_const(left, d), get_const(right, d));
+    let constant = right_const - left_const;
+
+    let result: Result<(u32, u32, i32), String> = match (clocks_left, clocks_right) {
+        (1, 1) => {
+            let (c1, c2) = (
+                get_clock_val(left, d, 1, false)?.0,
+                get_clock_val(right, d, 1, false)?.0,
+            );
+            combine_clocks(c1, c2, constant, true)
+        }
+        (2, 0) => {
+            let (c1, c2) = get_clock_val(left, d, 2, false)?;
+            combine_clocks(c1, c2.unwrap(), constant, false)
+        }
+        (0, 2) => {
+            let (mut c1, c2) = get_clock_val(right, d, 2, false)?;
+            let mut c2 = c2.unwrap();
+            c1.invert();
+            c2.invert();
+            combine_clocks(c1, c2, constant, false)
+        }
+        (1, 0) => {
+            let c = get_clock_val(left, d, 1, false)?.0;
+            if c.negated {
+                Ok((0, c.value, constant))
+            } else {
+                Ok((c.value, 0, constant))
+            }
+        }
+        (0, 1) => {
+            let c = get_clock_val(right, d, 1, false)?.0;
+            if !c.negated {
+                Ok((0, c.value, constant))
+            } else {
+                Ok((c.value, 0, constant))
+            }
+        }
+        _ => {
+            if left_const.is_negative() {
+                Ok((0, (-left_const) as u32, right_const))
+            } else {
+                Ok((left_const as u32, 0, right_const))
+            }
```

Comment:
This should never happen, it doesn't make sense to have a constant in the clock position of the (u32, u32, i32) tuple. The only thing that would make sense here is to panic or Err. Intuitively this is the case where two integers are compared fx. `2 <= 3`, but this is either true or false, and should be evaluated and handled before we ever get to call get_indices here https://github.com/Ecdar/Reveaal/blob/4bb1c52066a3cc8153da0391c67797d4ceb2d3bd/src/EdgeEval/constraint_applyer.rs#L71.

Should be changed to this as all possible cases are covered
```suggestion
        _ => unreachable!()
```